### DISCUSSION TITLE
Add a class to Navbar Item

### DISCUSSION THREAD

**Discussion by OmegaSquared** [quarto-dev/quarto-cli#2858] [Q&A] [answered]
Is it possible to add a class name in Yaml to a navbar item? 

, I have a drop-down link in my navbar that is password protected using my serverless backend. When I log in, I want to change the icon from a lock to a unlock. Changing the icon is easy in JS, but I can't reference the icon I would like altered without an id or class on the drop-down item. 

Give us slightly more control of the navbar, and Quatro would be better than any other R markdown format for the web. My current workaround has a floating login button, not on the navbar. It isn't pleasant. I want it in the navbar. 

Please and thank you! 

Aaron Soderstrom 
Omega Squared

**Comment by ggirton** [comment]
This is not an answer, but also a desire for slightly more navbar control.  I would like to add Rel=“me” after (or before)  the href attribute in the anchor tag for the navbar.    Maybe a new key in the yaml, like rel-me: “me”, or something likue that.  Thank you!

**Reply by mcanouil** [reply]
What this issue is about is not the same thing (even if related in some ways) to the "rel=me" attribute for link which is needed especially for Mastodon check.
For the first point, @dragonstyle mentioned elsewhere that more control on the navbar and generally more control over the layout is planned in the future. To note, classes could be added using a Lua filter.
For the second point, please refer to discussion #3207 (or #3298) and the associated issue #3212

**Comment by ggirton** [comment]
Mickaël, Thank you!

I noticed in 3207 a clarification on how to use aria-label, without quotes.  And I put that up on my quarto blog, so I hope it works, because — after that! — I realised it was just someone discussing what they wanted, and that the rel-me: true citation had quotes.  Maybe yaml is by convention without quotes, whereas in css or html we are used to >>quoting<< everything?

Anyway, it is true as you point out that the two issues in the original discussion are distinct.  I guess I never saw those other discussions which you cite because I was searching on “navbar” rather than on Mastodon.  Sorry to bother everyone 🍸😸

All best && cheers George  

www.minimumviablepython.com

links:
- icon: mastodon
aria-label: Mastodon
href: ***@***.***
rel: "me"

**Reply by cscheid** [reply]
We _just_ merged the ability to add `rel` attributes to navigation items, including the about page. So this should work in the v1.3 prerelease versions coming up. Thank @effigies for the feature and PR!

**Reply by mcanouil** [reply]
@ggirton no worries. I mostly linked to the proper discussion to help others find their way. 😉

**Reply by OmegaSquared** [reply]
Awesome thank you!

**Reply by OmegaSquared** [reply]
@cscheid post your answer, and I will mark it as the answer, thanks.

**Comment by cscheid** [comment, marked as answer]
We merged the ability to add rel attributes to navigation items, including the about page. So this should work in the https://github.com/quarto-dev/quarto-cli/releases/tag/v1.3.11 release.
